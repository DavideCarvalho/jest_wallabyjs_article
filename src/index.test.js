// @flow
const api = require('./index');
const axios = require('axios');

describe('index.js', () => {
  it('adds 1 + 2 to equal 3', () => {
    expect(api.somaDoisNumeros(1,2)).toMatchSnapshot();
  })
  
  it('expect to mock axios request', async () => {
    const spy = jest.spyOn(axios, 'get');
    spy.mockReturnValue('MockedValue');
    const apiResponse = await api.pegaPost(1);
    expect(apiResponse).toBe('MockedValue');
    expect(spy).toHaveBeenCalled();
  })
})

