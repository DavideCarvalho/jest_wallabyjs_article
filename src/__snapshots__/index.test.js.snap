// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.js adds 1 + 2 to equal 3 1`] = `3`;
